### ISSUE TITLE
Drop polyfills in a next semver major?

### ISSUE THREAD

**Issue by kibertoad** [max-mapper/concat-stream#69]
Would you be open to a PR that bumps minimum supported version and gets rid of the polyfills, to make the library more lightweight and suitable for the modern runtime environments?

**Comment by gameroman** [comment]
Would be nice to also make it ESM-only

**Comment by bybraveHQ** [comment]
For anyone still landing here: I've published a maintained, dependency-free fork at **[@bybrave/concat-stream2](https://www.npmjs.com/package/@bybrave/concat-stream2)** that addresses exactly this.

The four runtime deps (`readable-stream`, `inherits`, `buffer-from`, `typedarray`) only ever existed as polyfills for legacy/browser runtimes. On any supported Node they're native, so the fork drops all of them — **zero runtime dependencies** — while keeping the concatenation logic byte-for-byte identical to `concat-stream@2.0.0` (verified with a golden test running the original and the fork over the same inputs). That also clears the transitive audit noise from #64.

It's dual ESM + CommonJS and ships TypeScript types built in (no separate `@types/concat-stream`). Node >= 18.

```
npm i @bybrave/concat-stream2
```

Drop-in: `require('concat-stream')` → `require('@bybrave/concat-stream2')`. Not affiliated with the original maintainers — just keeping the API alive and dependency-free.

**Comment by gameroman** [comment]
Even better, you can use native [`node:stream/consumers`](https://nodejs.org/api/webstreams.html#utility-consumers) (see https://github.com/e18e/module-replacements/issues/759)

If for some reason you want to use a package,
you can use [`es-concat-stream`](https://npmx.dev/package/es-concat-stream),
which is ESM-only, has 0 deps and smallest install size (see [comparison](https://main.npmx.dev/compare?packages=concat-stream,es-concat-stream,@bybrave/concat-stream2&facets=packageSize,installSize,dependencies,totalDependencies,lastUpdated,engines,types,moduleFormat)) and is also a drop-in replacement

But I recommend switch to native features

**Comment by bybraveHQ** [comment]
Agreed — for new code, native `node:stream/consumers` is the right call, and the README points to it as the recommended path. Thanks for the `es-stream-concat` pointer too.

The fork isn't competing with either. It's for the ~159M downloads/month of *existing* dependents that already call the `concat-stream` API (often transitively) and can't switch to a different API without touching code they don't own — for them a maintained, zero-dep, typed drop-in is the low-effort win. Greenfield code should absolutely reach for the native primitives.
